src/jit/morph.cpp

```diff
@@ -4826,9 +4862,10 @@ GenTreePtr Compiler::fgMorphMultiregStructArg(GenTreePtr arg, fgArgTabEntryPtr f
 
                     // Create a new tree for 'arg'
                     //    replace the existing LDOBJ(ADDR(LCLVAR))
-                    //    with a LIST(LCLVAR-LO, LIST(LCLVAR-HI, nullptr))
+                    //    with a FIELD_LIST(LCLVAR-LO, FIELD_LIST(LCLVAR-HI, nullptr))
                     //
-                    newArg = gtNewAggregate(hiLclVar)->Prepend(this, loLclVar);
```

Comment:
I really don't like the previous approach where the fields are added backward. Once they are changed to be added in forward order, the `Prepend` is not needed. I could add an "Append" but personally I think this is more transparent.

---
In reply to: [79511593](https://github.com/dotnet/coreclr/pull/7252#discussion_r79511593) [](ancestors = 79511593)